src/sentry/tasks/symbolication.py

```diff
@@ -0,0 +1,339 @@
+
+error_logger = logging.getLogger("sentry.errors.events")
+info_logger = logging.getLogger("sentry.symbolication")
+
+# Is reprocessing on or off by default?
+REPROCESSING_DEFAULT = False
+
+SYMBOLICATOR_MAX_RETRY_AFTER: int = settings.SYMBOLICATOR_MAX_RETRY_AFTER
+
+
+class RetrySymbolication(Exception):
+    def __init__(self, retry_after: Optional[int] = None) -> None:
+        self.retry_after = retry_after
+
+
+def should_demote_symbolication(project_id: int) -> bool:
+    """
+    Determines whether a project's symbolication events should be pushed to the low priority queue.
+    """
+    always_lowpri = killswitch_matches_context(
+        "store.symbolicate-event-lpq-always",
+        {
+            "project_id": project_id,
+        },
+    )
+    never_lowpri = killswitch_matches_context(
+        "store.symbolicate-event-lpq-never",
+        {
+            "project_id": project_id,
+        },
+    )
+    return not never_lowpri and always_lowpri
+
+
+def submit_symbolicate(
+    from_reprocessing: bool,
+    cache_key: Optional[str],
+    event_id: Optional[str],
+    start_time: Optional[int],
+    data: Optional[Any],
+) -> None:
+    task = symbolicate_event_from_reprocessing if from_reprocessing else symbolicate_event
+    task.delay(cache_key=cache_key, start_time=start_time, event_id=event_id)
+
+
+def submit_symbolicate_low_priority(
+    from_reprocessing: bool,
+    cache_key: Optional[str],
+    event_id: Optional[str],
+    start_time: Optional[int],
+    data: Optional[Any],
+) -> None:
+    task = (
+        symbolicate_event_from_reprocessing_low_priority
+        if from_reprocessing
+        else symbolicate_event_low_priority
+    )
+    task.delay(cache_key=cache_key, start_time=start_time, event_id=event_id)
+
+
+def _do_symbolicate_event(
+    cache_key: Optional[str],
+    start_time: Optional[int],
+    event_id: Optional[str],
+    symbolicate_task: Any,
+    data: Optional[Any] = None,
+) -> None:
+    from sentry.lang.native.processing import get_symbolication_function
+    from sentry.tasks.store import _do_process_event, process_event, process_event_from_reprocessing
+
+    if data is None:
+        data = processing.event_processing_store.get(cache_key)
+
+    if data is None:
+        metrics.incr(
+            "events.failed", tags={"reason": "cache", "stage": "symbolicate"}, skip_internal=False
+        )
+        error_logger.error("symbolicate.failed.empty", extra={"cache_key": cache_key})
+        return
+
+    data = CanonicalKeyDict(data)
+
+    project_id = data["project"]
+    set_current_event_project(project_id)
+
+    event_id = data["event_id"]
+
+    from_reprocessing = symbolicate_task is symbolicate_event_from_reprocessing
+
+    def _continue_to_process_event() -> None:
+        process_task = process_event_from_reprocessing if from_reprocessing else process_event
+        # TODO: this uses a private "store" function. is there a way to not do this?
```

Comment:
nts: figure out an answer to this, namely "is this ok?" and how to make it ok if it isn't

also probably good to think about the the line that has been drawn by splitting all of the symbolicator stuff out into its own file: are there other places where it's using internal `store.py` stuff? is there symbolication logic in `store.py` that could and should be moved out into a method in this file? what are the low hanging fruit for those?

Reply:
I LEFT THIS ON THE WRONG PR SOMEHOW I'M SORRY